code/drasil-lang/lib/Language/Drasil/Expr/Lang.hs

```diff
@@ -229,17 +225,25 @@ vvvInfer ctx op l r = case (infer ctx l, infer ctx r) of
     (_       , Right re) -> Right re
     (Right le, _       ) -> Right le
 
+
 instance Typed Expr Space where
   check :: TypingContext Space -> Expr -> Space -> Either Space TypeError
   check = typeCheckByInfer
 
   infer :: TypingContext Space -> Expr -> Either Space TypeError
   infer cxt (Lit lit) = infer cxt lit
 
-  infer cxt (AssocA op exs) = allOfType cxt exs sp sp
-      $ "Associative arithmetic operation expects all operands to be of the same expected type (" ++ show sp ++ ")."
+  infer cxt (AssocA _ exs) = 
```

Comment:
If I use pattern matching I have to deal with the case of an empty list. Is that better than just using `head` on exs?

Reply:
Yes. I don't think the non-empty or single-item list will ever occur because of our smart constructors, but if it does, let's at least have a better error message than whatever `head` gives.